tests/components/alert/test_init.py

```diff
@@ -117,239 +92,237 @@ def async_toggle(hass, entity_id):
-        assert setup_component(self.hass, alert.DOMAIN, config)
-
-        self.hass.states.set(TEST_ENTITY, STATE_ON)
-        self.hass.block_till_done()
-        self.assertEqual(1, len(events))
-        last_event = events[-1]
-        self.assertEqual(last_event.data[notify.ATTR_TITLE], TEST_TITLE)
-
-    def test_sending_data_notification(self):
-        """Test notifications."""
-        events = self._setup_notify()
-
-        config = deepcopy(TEST_CONFIG)
-        config[alert.DOMAIN][NAME][alert.CONF_DATA] = TEST_DATA
-        assert setup_component(self.hass, alert.DOMAIN, config)
-
-        self.hass.states.set(TEST_ENTITY, STATE_ON)
-        self.hass.block_till_done()
-        self.assertEqual(1, len(events))
-        last_event = events[-1]
-        self.assertEqual(last_event.data[notify.ATTR_DATA], TEST_DATA)
-
-    def test_skipfirst(self):
-        """Test skipping first notification."""
-        config = deepcopy(TEST_CONFIG)
-        config[alert.DOMAIN][NAME][alert.CONF_SKIP_FIRST] = True
-        events = []
-
-        @callback
-        def record_event(event):
-            """Add recorded event to set."""
-            events.append(event)
-
-        self.hass.services.register(notify.DOMAIN, NOTIFIER, record_event)
-
-        assert setup_component(self.hass, alert.DOMAIN, config)
-        assert len(events) == 0
-
-        self.hass.states.set("sensor.test", STATE_ON)
-        self.hass.block_till_done()
-        assert len(events) == 0
-
-    def test_noack(self):
-        """Test no ack feature."""
-        entity = alert.Alert(self.hass, *TEST_NOACK)
-        self.hass.add_job(entity.begin_alerting)
-        self.hass.block_till_done()
-
-    def test_done_message_state_tracker_reset_on_cancel(self):
-        """Test that the done message is reset when canceled."""
-        entity = alert.Alert(self.hass, *TEST_NOACK)
-        entity._cancel = lambda *args: None
-        assert entity._send_done_message is False
-        entity._send_done_message = True
-        self.hass.add_job(entity.end_alerting)
-        self.hass.block_till_done()
-        assert entity._send_done_message is False
+@pytest.fixture
+def mock_notifier(hass):
+    """Mock for notifier."""
+    events = []
+
+    @callback
+    def record_event(event):
+        """Add recorded event to set."""
+        events.append(event)
+
+    hass.services.async_register(notify.DOMAIN, NOTIFIER, record_event)
+
+    return events
+
+
+async def test_is_on(hass):
+    """Test is_on method."""
+    hass.states.async_set(ENTITY_ID, STATE_ON)
+    await hass.async_block_till_done()
+    assert alert.is_on(hass, ENTITY_ID)
+    hass.states.async_set(ENTITY_ID, STATE_OFF)
+    await hass.async_block_till_done()
+    assert not alert.is_on(hass, ENTITY_ID)
+
+
+async def test_setup(hass):
+    """Test setup method."""
+    assert await async_setup_component(hass, alert.DOMAIN, TEST_CONFIG)
+    assert STATE_IDLE == hass.states.get(ENTITY_ID).state
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6356951)